sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -166,9 +166,6 @@ export class Receiver {
    */
   async renewMessageLock(lockTokenOrMessage: string | ServiceBusMessage): Promise<Date> {
     this._throwIfReceiverOrConnectionClosed();
-    if (this._receiveMode !== ReceiveMode.peekLock) {
-      throw new Error("The operation is only supported in 'PeekLock' receive mode.");
-    }
```

Comment:
AFAIK Renewal of message lock can only be done in PeekLock mode. Why are we removing this check here?